serde-generate/tests/ocaml_runtime.rs

```diff
@@ -0,0 +1,120 @@
+use serde_generate::{
+    ocaml, test_utils, test_utils::Runtime, CodeGeneratorConfig, SourceInstaller,
+};
+use std::{fs::File, io::Write, process::Command};
+use tempfile::tempdir;
+
+fn quote_bytes(bytes: &[u8]) -> String {
+    format!(
+        "\"{}\"",
+        bytes
+            .iter()
+            .map(|x| format!("\\{:03}", x))
+            .collect::<Vec<_>>()
+            .join("")
+    )
+}
+
+#[test]
+fn test_ocaml_bcs_runtime() {
+    test_ocaml_runtime(Runtime::Bcs);
+}
+
+#[test]
+fn test_ocaml_bincode_runtime() {
+    test_ocaml_runtime(Runtime::Bincode);
+}
+
+fn test_ocaml_runtime(runtime: Runtime) {
```

Comment:
Is it possible to avoid two tests `test_ocaml_runtime_on_simple_data` and `test_ocaml_runtime_on_supported_types` as in the other languages?
The second one is what you have already (I think). The first one is typically the only place where we actually test "semantic" interoperability between Rust and the other language.

Reply:
will do